fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgres/PostgresVacuumSettingDAO.java

```diff
@@ -49,70 +60,86 @@ public PostgresVacuumSettingDAO(String schema, String tableName, int vacuumCostL
+                        for (String setting : sar) {
+                            String[] kv = setting.split("=");
+                            if ("autovacuum_vacuum_scale_factor".equals(kv[0]) && vacuumScaleFactor == Double.parseDouble(kv[1])) {
+                                processSettings1 = false;
+                            } else if ("autovacuum_vacuum_threshold".equals(kv[0]) && vacuumThreshold == Integer.parseInt(kv[1])) {
+                                processSettings2 = false;
+                            } else if ("autovacuum_vacuum_cost_limit".equals(kv[0]) && vacuumCostLimit == Integer.parseInt(kv[1])) {
+                                processSettings3 = false;
+                            }
                         }
                     }
                 }
-            }
-        } catch (SQLException x) {
-            throw translator.translate(x);
-        }
-
-        if (processSettings1a || processSettings1b) {
-            // -- Lower the trigger threshold for starting work
-            // alter table fhirdata.logical_resources SET (autovacuum_vacuum_scale_factor = 0.01,
-            // autovacuum_vacuum_threshold=1000);
-            final String statement1 = "alter table " + schema + "." + tableName
-                    + " SET (autovacuum_vacuum_scale_factor = " + vacuumScaleFactor + ", autovacuum_vacuum_threshold=" + vacuumThreshold + ")";
-
-            LOG.fine(() -> "Updating the table " + schema + "." + tableName + " [" + vacuumScaleFactor + "," + vacuumThreshold + "]");
-            try (PreparedStatement ps = c.prepareStatement(statement1)) {
-                ps.execute();
             } catch (SQLException x) {
                 throw translator.translate(x);
             }
-        }
 
-        if (processSettings2) {
-            // -- Increase the amount of work vacuuming completes before taking a breather (default is typically 200)
-            // alter table fhirdata.logical_resources SET (autovacuum_vacuum_cost_limit=2000);
-            final String statement2 = "alter table " + schema + "." + tableName + " SET (autovacuum_vacuum_cost_limit=" + vacuumCostLimit + ")";
-            LOG.fine(() -> "Updating the table autovacuum_vacuum_cost_limit " + schema + "." + tableName + " [" + vacuumCostLimit + "]");
-            try (PreparedStatement ps = c.prepareStatement(statement2)) {
-                ps.execute();
-            } catch (SQLException x) {
-                throw translator.translate(x);
+            // Build the SQL
+            StringBuilder builder = new StringBuilder();
+            builder.append("alter table ")
+                .append(schema).append(".").append(tableName)
+                .append(" SET ( ")
+                .append(
+                    generateWiths(processSettings1, processSettings2, processSettings3).stream()
+                        .map(with -> with.buildWithComponent())
+                        .collect(Collectors.joining(",")))
+                .append(" )");
```

Comment:
not a big deal, but could move this block of code inside the check on line 109

Reply:
I moved the code